docs/_docs/31-fasterkv-revivification.md

```diff
@@ -0,0 +1,68 @@
+
+### Design of FreeList
+The FreeList hierarchy consists of:
+- The FreeRecordPool, which maintains the bins, deciding which bin should be used for Enqueue and Dequeue.
+- Multiple FreeRecordBins, including one for oversize (where the length is too long to fit into the FreeRecordHeader)
+  - Each FreeRecordBin is partitioned (currently the number of partitions is ProcessorCount / 2) and the ManagedThreadId is used to decide which partition to start on for both enqueueing and dequeueing. If the initial partition cannot accommodate the operation (it is full for enqueue or empty for dequeue), the next partition is done, wrapping around until all partitions have been examined.
+- FreeRecords, which are syntactic sugar over a `long` value that contains the Address and Size.
+  - The Address is stored similarly to `RecordInfo.PreviousAddress`, with the same number of bits
+  - The Size is shifted above the address, using the remaining bits. Thus it is limited to 16 bits; any free records over that size go into the oversize bin.
+    - The smallest bin is of size 16, allowing reuse of records large enough to hold an integer key and value. 
+
+#### Fixed vs. VarLen
+For non-variable-length types, the record size is fixed, so the FreeRecordPool has only the one bin for it. Otherwise, it has the full range of variable-length bins.
+
+#### Enqueueing
+Bins hold records up to their maxium size, which is a power of 2. Enqueuing will enqueue a record into the "smallest" bin it fits into.
+
+#### Dequeueing
+Records stored in a bin have lengths that are *between* the lowest and highest sizes of a bin, *not* the highest size of the bin. Therefore, for performance we have to retrieve from the next-highest bin; e.g. 48 will come from the [64-127] bin because the [32-63] bin might not have anything larger than 42, so we can't satisfy the request and it would take too much time to find that out before moving to the larger bin.
+
+## Revivifying a record
+The FreeRecordList is used for:
+- An Upsert, RMW, or Delete that adds a new record
+- A Pending Read or RMW that does CopyToTail (Revivification is not used for the ReadCache).
+
+If there is a record available, it is reused; otherwise, the usual BlockAllocate is done.
+
+### Record Length Management
+There are two lengths we are concerned with: The FullValueLength (allocated space), and the UsedValueLength within that that the application has used. These are properties of the new `UpdateInfo` structure passed by reference to `IFunctions` update callbacks for Upsert, RMW, and Delete.
+
+The used space is for a new record is equal to the full allocation space; by default, "all space is used". Note that SpanByte actually initializes itself to include space for the entire record. However, other data types may not do this.
```

Comment:
I do not get the comment about SpanByte here. Maybe clarify.

Reply:
Changed to:
The `VariableLenghtBlittableAllocator.GetValue` overload with (physicalAddress, physicalAddress + allocatedSize) that calls the default SpanByte implementation of `IVariableLengthStructureSettings` actually initializes the value space to be a valid SpanByte that includes the entire space (up to the maximum number of elements that can fit). However, other data types may not do this.